azure-functions-powershell-library/test/E2E/Get-FunctionsMetadata.Tests.ps1

```diff
@@ -0,0 +1,104 @@
+﻿using module AzureFunctions.PowerShell.SDK
+
+Describe 'Empty App' {
+    BeforeAll {
+        $metadata = Get-FunctionsMetadata ((Get-Location).Path + "\apps\empty-app")
+        $metadataObject = $metadata | ConvertFrom-Json
+
+        Import-Module ".\helpers.psm1" -force
+    }
+    It 'Should return 0 functions' {
+        $metadataObject.Count | Should -Be 0
+    }
+}
+
+Describe 'Poorly formatted .ps1 file' {
+    It 'Should throw an error with the file name' {
+        { Get-FunctionsMetadata ((Get-Location).Path + "\apps\poorly-formatted") } 
+            | Should -Throw '*apps\poorly-formatted\function.ps1*'
+    }
+}
+
+Describe 'Hybrid or legacy app' {
+    It 'Should throw an error when legacy functions are present' {
+        { Get-FunctionsMetadata ((Get-Location).Path + "\apps\hybrid-model") } 
+            | Should -Throw
+    }
+}
+
+Describe 'Single Ps1 Function App' {
+    BeforeAll {
+        $metadata = Get-FunctionsMetadata ((Get-Location).Path + "\apps\single-ps1app")
+        $metadataObject = $metadata | ConvertFrom-Json
+
+        Import-Module ".\helpers.psm1" -force
+    }
+    It 'Should return 1 function' {
+        $metadataObject.Count | Should -Be 1
+    }
+
+    It 'Functions name should be correct' {
```

Comment:
For the test case name, what about `'Functions name should include <expectedname>'`?

Reply:
Done

Reply:
Improved further, using parameterized tests